packages/myst-transforms/src/blocks.ts

```diff
@@ -80,24 +80,22 @@ export function blockMetadataTransform(mdast: GenericParent, file: VFile) {
       }
     }
     if (block.identifier) {
-      const codeChildren = selectAll('code', block) as Code[];
-      codeChildren.forEach((child, index) => {
-        if (child.identifier) return;
-        if (codeChildren.length === 1) {
-          child.identifier = `${block.identifier}-code`;
-        } else {
-          child.identifier = `${block.identifier}-code-${index}`;
-        }
-      });
-      const outputChildren = selectAll('output', block) as GenericNode[];
-      outputChildren.forEach((child, index) => {
-        if (child.identifier) return;
-        if (outputChildren.length === 1) {
-          child.identifier = `${block.identifier}-output`;
-        } else {
-          child.identifier = `${block.identifier}-output-${index}`;
-        }
-      });
+      const codeNode = selectAll('code', block) as any as Code | null;
+      if (codeNode !== null && !codeNode.identifier) {
+        codeNode.identifier = `${block.identifier}-code`;
+      }
+      const outputsNode = select('outputs', block) as GenericNode | undefined;
+      if (outputsNode !== undefined && !outputsNode.identifier) {
+        // Label outputs node
+        outputsNode.identifier = `${block.identifier}-output`;
```

Comment:
I think that these ID changes may need to be taken into account in the upgrade/downgrade script for thebe to continue working.